mpmath/ctx_mp_python.py

```diff
@@ -365,8 +365,8 @@ class behaves fairly similarly to Python's complex type.
 
     def __new__(cls, real=0, imag=0):
         s = object.__new__(cls)
-        if isinstance(real, complex_types):
-            real, imag = real.real, real.imag
+        if isinstance(real, complex_types) or isinstance(real, numbers.Complex):
+            real, imag = real.real, (real.imag if imag == 0 else imag)
```

Comment:
What is the logic here?

I think the extra comparison with 0 is going to needlessly slow down the common case. Perhaps make a separate if-case for the isinstance(real, numbers.Complex) case if this is needed.

The new isinstance call will also slow down the common case where the input is real, but that perhaps hard to avoid if we want to support this feature.